files/setup-02-proxy.sh

```diff
@@ -23,28 +23,42 @@ if [ -n "${HTTP_PROXY}" ] || [ -n "${HTTPS_PROXY}" ]; then
     fi
 
     if [ ! -z "${HTTP_PROXY}" ]; then
-        if [ $YES_CREDS -eq 1 ]; then
-            HTTP_PROXY_URL="http://${PROXY_USERNAME}:${PROXY_PASSWORD}@${HTTP_PROXY}"
-        else
-            HTTP_PROXY_URL="http://${HTTP_PROXY}"
-        fi
-        echo "HTTP_PROXY=\"${HTTP_PROXY_URL}\"" >> ${PROXY_CONF}
-        cat > ${DOCKER_PROXY}/http-proxy.conf << __HTTP_DOCKER_PROXY__
+        a=(`printf '%s\n' "${HTTP_PROXY//\:\/\//$'\n'}"`)
+        if [ ${#a[*]} -eq 2 ]; then
+            HTTP_PROXY_PREFIX=${a[0]}
+            HTTP_PROXY=${a[1]}
+            if [ $YES_CREDS -eq 1 ]; then
+                HTTP_PROXY_URL="${HTTP_PROXY_PREFIX}://${PROXY_USERNAME}:${PROXY_PASSWORD}@${HTTP_PROXY}"
+            else
+                HTTP_PROXY_URL="${HTTP_PROXY_PREFIX}://${HTTP_PROXY}"
+            fi
+            echo "HTTP_PROXY=\"${HTTP_PROXY_URL}\"" >> ${PROXY_CONF}
+            cat > ${DOCKER_PROXY}/http-proxy.conf << __HTTP_DOCKER_PROXY__
 [Service]
 Environment="HTTP_PROXY=${HTTP_PROXY_URL}" "NO_PROXY=${NO_PROXY}"
 __HTTP_DOCKER_PROXY__
+        else
+	    echo -e "\e[91mInvalid HTTP Proxy URL supplied" > /dev/console
+        fi
     fi
 
     if [ ! -z "${HTTPS_PROXY}" ]; then
-        if [ $YES_CREDS -eq 1 ]; then
-            HTTPS_PROXY_URL="https://${PROXY_USERNAME}:${PROXY_PASSWORD}@${HTTPS_PROXY}"
-        else
-            HTTPS_PROXY_URL="https://${HTTPS_PROXY}"
-        fi
-        echo "HTTPS_PROXY=\"${HTTPS_PROXY_URL}\"" >> ${PROXY_CONF}
-        cat > ${DOCKER_PROXY}/https-proxy.conf << __HTTPS_DOCKER_PROXY__
+        a=(`printf '%s\n' "${HTTPS_PROXY//\:\/\//$'\n'}"`)
+        if [ ${#a[*]} -eq 2 ]; then
+            HTTPS_PROXY_PREFIX=${a[0]}
+            HTTPS_PROXY=${a[1]}
```

Comment:
For ease of debugging, would you mind creating a brand new variable like `HTTPS_PROXY_SERVER_PORT` rather than reusing `HTTPS_PROXY`? I had to take a closer look at the code to see what was happening and was confused on why HTTPS_PROXY was being redefined.